common/Fabric-Install-Utilities.psm1

```diff
@@ -585,6 +585,55 @@ function Write-Console($message){
     Write-Host $message -ForegroundColor Gray
 }
 
+function Test-DiscoveryHasBuildVersion($discoveryUrl, $credential) {
+    $response = [xml](Invoke-WebRequest -Method Get -Uri "$discoveryUrl/`$metadata" -Credential $credential -ContentType "application/xml")
+	
+    return $response.Edmx.DataServices.Schema.EntityType.Property.Name -contains 'BuildNumber'
+}
+
+function Add-DiscoveryRegistration($discoveryUrl, $credential, $discoveryPostBody) {
+    $hasVersion = Test-DiscoveryHasBuildVersion $discoveryUrl $credential
+
+    if($hasVersion) {
```

Comment:
You could reduce the duplicated code here by first setting all the common properties on the registration body. Then if the $hasVersion variable is true, then have a statement to set `$registrationBody.BuildNumber = $discoveryPostBody.buildVersion`.

Reply:
Oh neat, will do that